probe-rs/src/architecture/arm/ap/memory_ap/mock.rs

```diff
@@ -34,7 +36,7 @@ impl MockMemoryAp {
 }
 
 impl CommunicationInterface for MockMemoryAp {
-    fn flush(&mut self) -> Result<(), DebugProbeError> {
+    fn flush(&mut self) -> Result<(), ArmError> {
```

Comment:
This is a generic `CommunicationInterface` (although only used for ARM atm afaik), so I am not sure we should return an `ArmError`.

Reply:
Returning a generic error is inconvenient, because it's used in ARM specific code. But having a ARM specific error in generic code is also not optimal, so I'd propose making a new ARM specific variant of this interface. Sounds good?

Reply:
Sounds good :)